drivers/overlay/overlay.go

```diff
@@ -1200,8 +1252,26 @@ func (d *Driver) ApplyDiff(id, parent string, options graphdriver.ApplyDiffOpts)
 
 func (d *Driver) getDiffPath(id string) string {
 	dir := d.dir(id)
+	diffPath := path.Join(dir, "diff")
+	if ld, err := os.Readlink(diffPath); err == nil {
+		// `diff` of additional layer store is symlink to the actual `diff` directory.
+		diffPath = ld
+	}
+	return diffPath
+}
 
-	return path.Join(dir, "diff")
+func (d *Driver) getLowerDiffPaths(id string) ([]string, error) {
+	layers, err := d.getLowerDirs(id)
+	if err != nil {
+		return nil, err
+	}
+	for i, l := range layers {
+		if ld, err := os.Readlink(l); err == nil {
+			// `diff` of additional layer store is symlink to the actual `diff` directory.
+			layers[i] = ld
+		}
+	}
```

Comment:
same here, how would we handle relative symlinks?

Reply:
fixed to allow only absolute paths.